test/test-network/systemd-networkd-tests.py

```diff
@@ -1886,6 +1887,21 @@ def test_address_dad(self):
         print(output)
         self.assertNotRegex(output, '192.168.100.10/24')
 
+    def test_address_peer_ipv4(self):
+        # test for issue #17304
+        copy_unit_to_networkd_unit_path('25-address-peer-ipv4.network', '12-dummy.netdev')
+
+        for trial in range(2):
+            if trial == 0:
+                start_networkd()
+            else:
+                restart_networkd()
+
+            self.wait_online(['dummy98:routable'])
+
+            output = check_output('ip -4 address show dev dummy98')
+            self.assertRegex(output, 'inet 100.64.0.1 peer 100.64.0.2/32 scope global')
```

Comment:
Dots are not escaped for regex pattern. I would prefer `assertIn` rather than `assertRegex`.

Reply:
Thanks. Now `assertIn()` is used in this test. I will convert remaining `assertRegex()` in other tests in another PR.